kafka-connect-s3/src/test/java/io/confluent/connect/s3/TopicPartitionWriterTest.java

```diff
@@ -900,6 +910,93 @@ public void testAddingS3ObjectTags() throws Exception{
     verifyTags(expectedTaggedFiles);
   }
 
+  @Test (expected = DataException.class)
+  public void testExceptionOnNullKeys() throws Exception {
+    SinkRecord faultyRecord = new SinkRecord(TOPIC, PARTITION, Schema.STRING_SCHEMA, null,
+        Schema.STRING_SCHEMA, "1", 1, 0L, TimestampType.NO_TIMESTAMP_TYPE, sampleHeaders());
+    writeRecordWithKeysAndHeaders(faultyRecord);
+  }
+
+  @Test (expected = DataException.class)
+  public void testExceptionOnEmptyHeaders() throws Exception {
+    SinkRecord faultyRecord = new SinkRecord(TOPIC, PARTITION, Schema.STRING_SCHEMA, "key",
+        Schema.STRING_SCHEMA, "1", 1, 0L, TimestampType.NO_TIMESTAMP_TYPE, Collections.emptyList());
```

Comment:
What's the difference between "1" and 1?
I'd suggest naming these variables before calling this constructor. 
value = "1" (?)
etc ...